### ISSUE TITLE
Training data

### ISSUE THREAD

**Issue by mortezakz** [airalcorn2/Solr-LTR#2]
This is a great tutorial - thank you!
Would you mind please insert a snippet of the training data? I want to see how the structure/schema should look like. I need to produce using my own scripts. 

Thanks!

**Comment by airalcorn2** [comment]
At what point in the tutorial are you referring to?

**Comment by prateekagarwal95** [comment]
@airalcorn2 You explain things very well - Thanks !!
 Is there any python library that we can use for this functionality other than using CURL something like haystack or pysolr but it doesn't support this.

**Comment by airalcorn2** [comment]
@prateekagarwal95 - [you can use the `requests` library.](https://stackoverflow.com/a/25491297/1316276)

**Comment by mortezakz** [comment]
@airalcorn2  sorry I somehow missed the notification. Thanks for the prompt reply. 
So... the code snippet in this section:
https://github.com/airalcorn2/Solr-LTR#ranknet
seems to be referring to a variable named "relevant", but it's not up there which makes running the script impossible (I haven't tried running the others, but from superficial look like overall the goal of the tutorial is not to run the snippets and replicate the results?),

Basically, I have a set of queries and answers for them (human judgment, mostly one result marked as relevant), and I want to know what the format of the results of feature extraction, RankNet as well as the model that I upload to Solr should look like, so that I can generate them using my own scripts. Makes sense?

**Comment by airalcorn2** [comment]
@mortezakz  - you need to generate a NumPy array `X` that has shape ((`RERANK` - 1) * `N`, 2 * `dim`) where `RERANK` is the number of documents returned by your initial ranking model (e.g., BM25), `N` is the number of queries that had the relevant document returned in the top `RERANK` results, and `dim` is the number of features you extract for a given query/document pair. The reason `X` has 2 * `dim` columns is because each row in `X` contains the features for both a relevant document (in `X[:, :dim]`) and an irrelevant document (in `X[:, dim:]`). Once you get to this point, the remaining Keras code should work, including the bit at the end that tells you how to save the model parameters and push them to Solr (it's worth noting that my implementation of RankNet has now been [merged into Solr](https://github.com/apache/lucene-solr/commit/c5938f79e540f81b6d61560d324b150a5efd7011), but parameters are loaded in a different way then what is described in the tutorial; I plan on updating that).

Learning to rank models are generally only applied to the top `RERANK` documents first retrieved using some other model (e.g., BM25). So, if you wanted to re-rank 50 documents with RankNet and each query only had one relevant document in the corpus, then `X` would contain 49 rows per query (one for each irrelevant document). Queries that did not return a relevant document in the top `RERANK` results using the first model will obviously not be included in the training set because there won't be any features for the relevant document.

**Comment by mortezakz** [comment]
@airalcorn2 Thank you very much! I'm planning on getting back to this in a month or so, I'll look for your updated version in case you get to it.
